23 add docker support

eyedol opened the pull request:
**- Summary**
- Adds a script in the scripts folder that creates a docker container, builds it and runs.
- Adds a folder called docker which contains docker template files.
- Adds a script in the scripts folder that creates hugulp required folders.
- Updates `README` with clear instructions on how to run the script.

It fixes #23 

**- Test plan**
Run command below to create a default docker container. Note, the script is already executable.
```bash
./scripts/create-docker-machine-and-run-it
```

jbrodriguez commented on scripts/create-docker-machine-and-run-it:
```diff
@@ -0,0 +1,159 @@
+  sed -i '' -e "s|\$HOME/[a-zA-Z0-9]*/$|\$HOME/$APP_NAME/|g" \
+    Dockerfile
+  # Replace website name with set value
+  sed -i '' -e "s|APP_NAME=website|APP_NAME=$APP_NAME|" \
```
Should docker/Dockerfile.template contain an ```APP_NAME=website``` line ?

eyedol replied:
It shouldn't contain that. Left over from a refactor

jbrodriguez commented on scripts/create-hugo-site:
```diff
@@ -0,0 +1,80 @@
+
+readonly PROGNAME=$(basename "$0")
+readonly ARGS="$@"
+APP_NAME=
```
Shouldn't this be ```APP_NAME=app``` ?

eyedol replied:
I'm forcing the user to provide that via the command line option. Thinking more about it, I'll make it `APP_NAME=app` and relax the app name option.

jbrodriguez commented:
Hi @eyedol, it looks great !!

I made two comments, just to make sure, I haven't done any Docker development in quite a while :)

If it's ok as it is, I'll merge the PR.

Thinking ahead, would you consider becoming a collaborator ?

I wouldn't mind moving the project over to a github organization (hugulp?), so that you and other people could get involved.

eyedol commented:
@jbrodriguez Thanks for the review. I have addressed your comments in my latest push. I squashed everything into a single commit to prevent clutter in the commit history.

Yeah, I'm happy to contribute whenever I can. Feel free to add me as a collaborator. Making it an org will be better I think. Easier to track people. Feel free to add me on there when you have it created.

jbrodriguez commented:
@eyedol awesome, thanks for the PR !

I'll work on the org next week.
